packages/application-components/src/components/modal-pages/internals/modal-page.js

```diff
@@ -30,41 +42,182 @@ const getDefaultParentSelector = () =>
+      justify-content: center;
+      &::after {
+        content: '';
+        position: absolute;
+        height: 35px;
+        width: 48px;
+        top: 0;
+        right: 0;
+      }
+    `}
+  >
+    <CloseIcon size="medium" {...props} />
+  </span>
 );
+LargeCloseIcon.displayName = 'LargeCloseIcon';
+
+const ModalPageTopBar = injectIntl(props => (
+  <div
+    css={css`
+      position: relative;
+      display: flex;
+      align-items: center;
+      justify-content: space-between;
+      padding: ${customProperties.spacingS} ${customProperties.spacingM};
+      background-color: ${props.color === 'neutral'
+        ? customProperties.colorNeutral95
+        : customProperties.colorSurface};
+      border-bottom: 1px solid
+        ${props.color === 'neutral'
+          ? customProperties.colorSurface
+          : customProperties.colorNeutral};
+      & * + * {
+        margin-left: ${customProperties.spacingS};
+      }
+
+      /* FIXME: these "dirty" styles should be removed when the new Breadcrumbs component is implemented */
+      p {
+        font-size: 12px !important;
+      }
+
+      /*  specific selector for the svg of the FlatButton */
+      button > span > svg {
+        height: 12px !important;
+        width: 12px !important;
+      }
+    `}
+  >
+    <div
+      css={css`
+        display: flex;
+        overflow: hidden;
+      `}
+    >
+      <FlatButton
+        tone="primary"
+        label={
+          typeof props.previousPathLabel === 'string'
+            ? props.previousPathLabel
+            : props.intl.formatMessage(props.previousPathLabel)
+        }
+        icon={<AngleLeftIcon size="medium" theme="green" />}
+        onClick={props.onClose}
+      />
+      {props.currentPathLabel && (
+        <React.Fragment>
+          <Text.Detail isInline>/</Text.Detail>
+          <Text.Detail title={props.currentPathLabel} isInline truncate>
+            {props.currentPathLabel}
+          </Text.Detail>
+        </React.Fragment>
+      )}
+    </div>
+    {props.onClose && (
+      <SecondaryIconButton
+        label={props.intl.formatMessage(messages.close)}
+        onClick={props.onClose}
+        icon={<LargeCloseIcon />}
+      />
+    )}
+  </div>
+));
+ModalPageTopBar.displayName = 'ModalPageTopBar';
+ModalPageTopBar.propTypes = {
+  color: PropTypes.oneOf(['surface', 'neutral']),
+  onClose: PropTypes.func.isRequired,
+  currentPathLabel: PropTypes.string,
+  previousPathLabel: PropTypes.oneOfType([
+    PropTypes.string,
+    // default intl message
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      defaultMessage: PropTypes.string.isRequired,
+    }),
+  ]).isRequired,
+  // injected
+  intl: PropTypes.shape({
+    formatMessage: PropTypes.func.isRequired,
+  }),
+};
+ModalPageTopBar.defaultProps = {
+  color: 'surface',
+  previousPathLabel: messages.back,
+};
+
+const ModalPage = props => {
+  const [isOpen, setIsOpen] = React.useState(props.isOpen);
```

Comment:
Heads up: in theory the app-kit does not necessarily support hooks. We plan to do that in the v14 release (#748).
Maybe we should keep this on hold until then, if we're going to use hooks here?

Reply:
Ah I see.
Well, I guess it's okay to put this on hold and merge on the v14 release.

The alternative would be refactoring the ModalPage into a class component to use the traditional lifecycle methods... only to refactor them back on v14, so it's not worth it IMHO.

Reply:
Yeah I think it's not a big deal at the moment, and it shouldn't stop migrating the mc-fe to the new modal pages.

Reply:
Maybe we can start a `v14` branch already, then we can point this PR to that branch

Reply:
I think, for simplicity, I'd just not use hooks. Then in a consistent fashion we can refactor things to hooks in the future.